arviz/plots/backends/matplotlib/khatplot.py

```diff
@@ -105,6 +106,14 @@ def plot_khat(
         rgba_c[:, 3] = alphas
         rgba_c = vectorized_to_hex(rgba_c)
         kwargs["c"] = rgba_c
+    else:
+        if isinstance(c_kwarg, str):
+            if c_kwarg in dims:
+                colors, color_mapping = color_from_dim(khats, c_kwarg)
+            else:
+                legend = False
+        else:
+            legend = False
```

Comment:
I would not add that. I think `c` should always behave _exactly_ like the argument in `matplotlb.scatter` if they want any of the other features, users should use `color`